File: Printly-backend/app/schemas/invoice.py
from pydantic import BaseModel, Field, ConfigDict
from uuid import UUID
from datetime import datetime
from decimal import Decimal


class InvoiceResponse(BaseModel):
    id: UUID
    invoice_number: str
    order_id: UUID
    customer_id: UUID | None
    customer_name: str | None
    total_amount: Decimal
    paid_amount: Decimal
    notes: str | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class InvoicesRequest(BaseModel):
    customer_id: UUID | None = None
    order_id: UUID | None = None
    offset: int = 0
    limit: int = 10
    order_by: str = "created_at"
    order_dir: str = "desc"


class InvoiceListResponse(BaseModel):
    total: int
    items: list[InvoiceResponse]


class InvoiceDetailResponse(BaseModel):
    id: UUID
    invoice_number: str
    order_id: UUID
    order_number: str
    customer_id: UUID | None
    customer_name: str | None
    total_amount: Decimal
    paid_amount: Decimal
    notes: str | None
    created_at: datetime
    items: list[dict] = []
    payments: list[dict] = []

    model_config = ConfigDict(from_attributes=True)

File: Printly-frontend/src/types/invoice.ts
// ==================== Invoice Types ====================

export interface InvoiceResponse {
	id: string;
	invoice_number: string;
	order_id: string;
	customer_id: string | null;
	customer_name: string | null;
	total_amount: number;
	paid_amount: number;
	notes: string | null;
	created_at: string;
}

export interface InvoiceDetailResponse extends InvoiceResponse {
	order_number: string;
	items: InvoiceItem[];
	payments: InvoicePayment[];
}

export interface InvoiceItem {
	book_title: string;
	copies: number;
	unit_price: number;
	subtotal: number;
}

export interface InvoicePayment {
	id: string;
	amount: number;
	payment_method: string;
	reference: string | null;
	created_at: string;
}

export interface InvoicesRequest {
	customer_id?: string;
	order_id?: string;
	offset?: number;
	limit?: number;
	order_by?: string;
	order_dir?: string;
}

export interface InvoiceListResponse {
	total: number;
	items: InvoiceResponse[];
}
